### PR TITLE
Fix #1196: Changes the UI of Skill Listing, Skill details activity

### LINKED ISSUE #1196: A better UI for Skill Listing and skill details page
**Actual Behaviour**

The current UI for skill details page and skill listing page is good but don't follow all Material UI guidelines which can certainly help to engage the user even more.

**Expected Behaviour**

The various activities must follow Material UI guidelines and also there must be certain tweaks in the UI of the various pages.

**Steps to reproduce it**

We can just change the various XML files related to the various fragments and activities.

**Screenshots of the issue**
Previous Version:
susi_searchiconerror

New Version:
skilldetailsuichange
There would be many more UI changes once I complete the work

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by aniket-agarwal1999**
Fixes #1196 

**Changes**

Implemented Material UI criteria in the Skill details and Skill listing. Also changed the UI a bit following the UI of Google assistant. The UI changes are as follows:
1. Making the Share and Try button in a rounded form with a different layout altogether in the skill details page and also changing the font size according to the Material UI guidelines. 
2. Also changing the format in which a skill in a block in the skill listing page, which even includes rearranging the heading of the skill. 

**Screenshots of the change**

skilldetailsuichange

uichangeforskilllisting

### CODE REVIEW

**Comment by aniket-agarwal1999** [comment]
@chiragw15 Can you please review it?

**Comment by aniket-agarwal1999** [comment]
Can anyone please review this PR.

**Comment by aniket-agarwal1999** [comment]
@arundhati24 sure

**Comment by iamareebjamal** [comment]
@aniket-agarwal1999 Please resolve conflicts

**Comment by batbrain7** [comment]
@iamareebjamal can you please revert this PR, on merging this many of the inconsistencies and conflicts have risen and causing problem with the CI build failures

apart from the connectedCheck

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 1]
```diff
@@ -33,6 +33,9 @@
                     android:id="@+id/skill_detail_title"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
+                    android:text="@string/skill_name"
+                    android:textSize="20sp"
+                    android:layout_marginTop="4dp"
                     android:layout_marginTop="8dp"
                     android:background="@color/md_white_1000"
                     android:text="@string/skill_name"
```
Comment (batbrain7) [inline app/src/main/res/layout/fragment_skill_details.xml]: there is android:text="@string/skill_name" already present but it is added again

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 2]
```diff
@@ -92,6 +129,8 @@
             <TextView
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
+                android:text="@string/description"
+                android:textSize="14sp"
                 android:layout_marginTop="8dp"
                 android:text="@string/description"
                 android:textColor="@color/md_black_1000"
```
Comment (batbrain7) [inline app/src/main/res/layout/fragment_skill_details.xml]: android:text="@string/description" added twice

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 3]
```diff
@@ -101,6 +140,7 @@
                 android:id="@+id/skill_detail_description"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
+                android:textSize="14sp"
                 android:layout_marginTop="8dp"
                 android:text="@string/description_long"
                 android:textSize="16sp" />
```
Comment (batbrain7) [inline app/src/main/res/layout/fragment_skill_details.xml]: android:textSize added twice

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 4]
```diff
@@ -109,6 +149,8 @@
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
                 android:layout_marginTop="16dp"
+                android:textSize="14sp"
+                android:textColor="@color/md_black_1000"/>
                 android:text="@string/examples"
                 android:textColor="@color/md_black_1000"
```
Comment (batbrain7) [inline app/src/main/res/layout/fragment_skill_details.xml]: android:textColor added twice

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 5]
```diff
@@ -122,6 +164,8 @@
             <TextView
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
+                android:text="@string/rating"
+                android:textSize="14sp"
                 android:layout_marginTop="16dp"
                 android:text="@string/rating"
```
Comment (batbrain7) [inline app/src/main/res/layout/fragment_skill_details.xml]: android:text="@string/rating" added twice

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 6]
```diff
@@ -142,9 +186,37 @@
...
+                    <ImageView
+                        android:id="@+id/skill_detail_rating_thumbs_down"
+                        android:layout_marginRight="8dp"
+                        android:layout_gravity="center"
+                        android:src="@drawable/thumbs_down_solid"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content" />
                         android:layout_gravity="center_horizontal"
                         android:background="?attr/selectableItemBackground"
                         android:orientation="horizontal">
```
Comment (batbrain7) [inline app/src/main/res/layout/fragment_skill_details.xml]: Wrong addition  /> added up there but > is below it

File: app/src/main/res/layout/fragment_skill_details.xml [hunk 7]
```diff
@@ -214,6 +286,8 @@
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
                 android:layout_marginTop="16dp"
+                android:textSize="14sp"
+                android:textColor="@color/md_black_1000"/>
                 android:text="@string/content_type_dynamic"
                 android:textColor="@color/md_black_1000"
                 android:textSize="18dp" />
```
Comment (batbrain7) [inline app/src/main/res/layout/fragment_skill_details.xml]: same as the previous comment

File: app/src/main/res/layout/item_skill.xml [hunk 8]
```diff
@@ -13,7 +13,8 @@
         android:foreground="?attr/selectableItemBackground"
         android:layout_marginLeft="@dimen/margin_card"
         android:layout_marginTop="@dimen/message_card_margin_small"
-        app:cardBackgroundColor="@color/susi_message_bg">
+        app:cardBackgroundColor="@color/susi_message_bg"
+        app:cardElevation="4dp">
```
Comment (batbrain7) [inline app/src/main/res/layout/item_skill.xml]: same as the previous comment

File: app/src/main/res/values/dimens.xml [hunk 9]
```diff
@@ -39,6 +42,11 @@
     <dimen name="padding_chatbubble_large">15dp</dimen>
     <dimen name="padding_chatbubble_small">5dp</dimen>
     <dimen name="padding_img">40dp</dimen>
+    <dimen name="toolbar_logo_width">110dp</dimen>
+    <dimen name="toolbar_logo_height">22dp</dimen>
+    <dimen name="skill_title">12sp</dimen>
+    <dimen name="skill_description">14sp</dimen>
+    <dimen name="skill_min_height">60dp</dimen>
     <dimen name="padding_large">10dp</dimen>
```
Comment (batbrain7) [inline app/src/main/res/values/dimens.xml]: all these dimensions are already added in the app but are added again here

**Comment by iamareebjamal** [comment]
@aniket-agarwal1999 Please resend the PR

**Pull request merged**